### DISCUSSION TITLE
Feature request: Add coffe maker additional sensors

### DISCUSSION THREAD

**Discussion by jmadejek** [ekutner/home-connect-hass#153] [Ideas]
Please add some additional sensors for coffe makers (i saw it in HA homekit integration)

- information about milk still in the device
- information that you have to clean drip tray
- information that you have to fill the water

**Comment by ekutner** [comment]
The integration is automatically creating all the sensors that are exposed by the Home Connect API. Where exactly did you see these? If it's in the HomeConnect mobile app then they use a private API that is not publicly accessible.

**Reply by d0np3p3** [reply]
Some of these options are push messages, they should be push in the API too(not sure)
Like empty coffee, empty water.
Same with the dryers, empty fresh water.
Can I access the message from API via Homeassistent?
![Screenshot_20230203-163705.png](https://user-images.githubusercontent.com/93970411/216644031-445d8f9c-07d0-4a9b-bbb1-4b55e8ffa44f.png)

![Screenshot_20230203-163654.png](https://user-images.githubusercontent.com/93970411/216644068-15785099-2fd1-4e71-b3ab-2173f750a16f.png)

Here it shows the available function in Homebridge:

https://github.com/thoukydides/homebridge-homeconnect/wiki/Functionality

It is also find in the api doc

https://api-docs.home-connect.com/events?#availability-matrix

ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty | EVENT
ConsumerProducts.CoffeeMaker.Event.WaterTankEmpty | EVENT  
ConsumerProducts.CoffeeMaker.Event.DripTrayFull | EVENT


Thank you for Your nice work 👍

**Comment by ekutner** [comment]
Every time I checked, those events were not received through the API. If you think they are, then enable debug logging as indicated in the README, reproduce the state that would generate such an event, like running out of water in the coffee maker, and upload the log file here, with the time when the event happened.

**Reply by d0np3p3** [reply]
Hello i just got one event for Tray full,
I am looking for the events out of water, and coffe beans empty in the next time, I dont use milk
`2023-02-06 09:37:48.705 DEBUG (MainThread) [custom_components.home_connect_alt] Published event to Home Assistant event bus: BSH.Common.Status.OperationState = BSH.Common.EnumType.OperationState.Run
2023-02-06 09:37:48.706 DEBUG (MainThread) [home_connect_async.homeconnect] Received event from SSE stream: MessageEvent(type='EVENT', message='EVENT', data='{"haId":"SIEMENS-TI9575X1DE-68A40E378952","items":[{"displayvalue":" Event aktiv","handling":"none","key":"ConsumerProducts.CoffeeMaker.Event.DripTrayFull","level":"alert","name":"Tropfschalen voll","timestamp":1675672667,"value":"BSH.Common.EnumType.EventPresentState.Present"}]}', origin='https://api.home-connect.com', last_event_id='SIEMENS-TI9575X1DE-68A40E378952')
2023-02-06 09:37:48.706 DEBUG (MainThread) [home_connect_async.callback_registery] Broadcasting event: ConsumerProducts.CoffeeMaker.Event.DripTrayFull = BSH.Common.EnumType.EventPresentState.Present
2023-02-06 09:37:48.707 DEBUG (MainThread) [custom_components.home_connect_alt] Published event to Home Assistant event bus: ConsumerProducts.CoffeeMaker.Event.DripTrayFull = BSH.Common.EnumType.EventPresentState.Present`

**Comment by ekutner** [comment]
So as that log indicates, this event is already published to the Home Assistant event bus. You can listen to it and react accordingly.

**Reply by d0np3p3** [reply]
👍 I thought that this could be visible in the webif, I will check out the other events on the log.
And it could be cleared when a manuel 
Maybe to trigger a Push message.
I also thought about a state: 
Clean Tray / Fill Water which is cleared by BSH.Common.Status.LocalControlActive

**Reply by ekutner** [reply]
The integration is totally generic, it doesn't even know which events exist, and since the event is not provided as a state from the API it can't be made into a sensor in HA. The best that can be done is what it does now, which is to publish the event and let you handle it however you like in HA. That is not going to change.

**Reply by d0np3p3** [reply]
Thats fine, so we already can use the Events, for Information use, here is the event for empty beans:
`2023-02-08 12:21:45.349 DEBUG (MainThread) [home_connect_async.homeconnect] Received event from SSE stream: MessageEvent(type='EVENT', message='EVENT', data='{"haId":"SIEMENS-TI9575X1DE-68A40E378952","items":[{"displayvalue":" Event aktiv","handling":"none","key":"ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty","level":"warning","name":"Bohnenbehälter leer","timestamp":1675855304,"value":"BSH.Common.EnumType.EventPresentState.Present"}]}', origin='https://api.home-connect.com', last_event_id='SIEMENS-TI9575X1DE-68A40E378952')
`

**Comment by d0np3p3** [comment]
It took me some time... 
Example to use Empty Water Tank in a Automation (send a Meassage to mobile / Tv etc...) 
I am still lokking to make it a state or sensor (which could be cleared with the "local usage" event
`
platform: event
event_type:
  - home_connect_alt_event
event_data:
  device_id: "Your ID (look dev ->events) "
  key: ConsumerProducts.CoffeeMaker.Event.WaterTankEmpty
  value: BSH.Common.EnumType.EventPresentState`

Or 
ConsumerProducts.CoffeeMaker.Event.BeanContainerEmpty
ConsumerProducts.CoffeeMaker.Event.DripTrayFull
Still looking for the Milk one... ConsumerProducts.CoffeeMaker.Event.???MILK???

**Comment by d0np3p3** [comment]
I think this could be closed and maybe we could post the information how to use Events to the readme or discussion (collect all possible events?) 
@ekutner
